doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -143,12 +139,30 @@ list of assertion commands is given in :ref:`Assertions`. The command
 
    .. seealso:: :cmd:`Proof with`
 
-.. cmd:: Proof using {+ @ident }
+.. cmd:: Proof using @section_subset_expr {? with @ltac_expr }
 
-   This command applies in proof editing mode. It declares the set of
+   .. insertprodn section_subset_expr ssexpr0
+
+   .. prodn::
+      section_subset_expr ::= {* @starredidentref }
+      | @ssexpr
+      starredidentref ::= @ident {? * }
+      | Type {? * }
+      ssexpr ::= - @ssexpr50
+      | @ssexpr50
+      ssexpr50 ::= @ssexpr0 - @ssexpr0
+      | @ssexpr0 + @ssexpr0
+      | @ssexpr0
+      ssexpr0 ::= @starredidentref
+      | ( {* @starredidentref } ) {? * }
+      | ( @ssexpr ) {? * }
```

Comment:
Here's a proposal for an equivalent but simpler grammar, with some light renamings to make the reading better:
```
   .. prodn::
      section_subset_expr ::= {* @starred_ident_ref }
      | {? - } @section_subset_expr50
      section_subset_expr50 ::= @section_subset_expr0 - @section_subset_expr0
      | @section_subset_expr0 + @section_subset_expr0
      | @section_subset_expr0
      section_subset_expr0 ::= @starred_ident_ref
      | ( @section_subset_expr ) {? * }
      starred_ident_ref ::= @ident {? * }
      | Type {? * }
```

Reply:
Got it.  How about using `section_var_expr` for the nonterminals since the goal is to produce a set of idents.  Slightly shorter name. 
 And `All` is not a proper subset.

Reply:
Sure! By not a proper subset, you mean not a strict subset? Indeed, in French we tend to consider things as not strict by default but it's not the same in English.

Reply:
"proper subset" is the term I'm familiar with.  I don't recall hearing "strict subset".  This reference lists both: https://en.wikipedia.org/wiki/Subset.  "A is a proper (or strict) subset of B" suggests that that "proper subset" is more common.

> Indeed, in French we tend to consider things as not strict by default but it's not the same in English.

Meaning in a mathematical context?  Not quite sure how this relates to English.

I switched the grammar to "section_var_expr"

Reply:
Yes, in mathematical text. I mean this is a common ambiguity in English written by French people because of the different conventions.

Reply:
Ah no, "A is a subset of B" could mean they are equal; I don't think the terminology is the same in English.  I can't remember exactly why I commented "And `All` is not a proper subset."

Reply:
OK thanks for clarifying.